docs/product/components/tables.html

```diff
@@ -631,6 +633,76 @@
             </table>
         </div>
     </div>
+
+    {% header h3 | JavaScript sorting %}
+
+    <p class="stacks-copy">Stacks provides built-in functionality for letting the user sort a table by the values in a column through clicking the column header. This requires the complete data to already exist in the table (e.g. it is not going to work if the table is paged and requires a call to the server to update data on sorting). See the <a href="{{ "/product/guidelines/javascript" | relative_url }}">JavaScript introduction</a> for general information about JS in Stacks.</p>
+
+    <p class="stacks-copy">To make your table user-sortable, do the following:</p>
+
+    <ol class="stacks-list">
+        <li>Style the table as sortable as explained in the section above.</li>
+        <li>Set <code class="stacks-code">data-controller="s-table"</code> on the <code class="stacks-code">&lt;table&gt;</code> element.</li>
+        <li>Set <code class="stacks-code">data-target="s-table.column"</code> and <code class="stacks-code">data-action="click->s-table#sort"</code> on each of the <code class="stacks-code">&lt;th&gt;</code> elements that control sorting.</li>
+        <li>
+            Add the three icons for showing <i>ascending sort</i>, <i>descending sort</i>, and <i>unsorted</i> to each of these header cells, hiding the first two with a <code class="stacks-code">d-none</code> class. Add the <code class="stacks-code">js-sorting-indicator</code> class to each of the icons, and add <code class="stacks-code">js-sorting-indicator-asc</code>, <code class="stacks-code">js-sorting-indicator-desc</code>, and <code class="stacks-code">js-sorting-indicator-none</code> to the appropriate icon.
+
+            <div class="grid ai-start s-notice s-notice__info mt16">
+                <div class="grid--cell mr8">{% icon Alert %}</div>
+                <div class="grid--cell lh-xl">
+                    <b>Note:</b> Using <code class="stacks-code bg-powder-200">js-…</code> classes is not really the optimal way of doing this, and will probably replaced with something better eventually. When that happens, the <code class="stacks-code bg-powder-200">js-…</code> mechanism will be deprecated but continue to be supported for a while, so you have ample time to update things.
+                </div>
+            </div>
+        </li>
+    </ol>
+
+    <p class="stacks-copy">By default, the data is sorted by the content of the cell. If you need to use a different value, for example because your cell contains a human-readable date, add a <code class="stacks-code">data-s-table-sort-val</code> attribute to the cell.</p>
+
+    <p class="stacks-copy">If a column contains any data that is not an integer, the data will be sorted lexicographically. Otherwise it will be sorted numerically, with empty cells being considered the lowest number.</p>
```

Comment:
There's a subtle difference (at least in how developers tend to use the words). Alphabetically means "a" comes before "Z", but lexicographically "a" comes after "Z" because lowercase letters have higher codepoints.